### ISSUE TITLE
Stripe Payment Gateway Integration

### ISSUE THREAD

**Issue by oreveha** [forcedotcom/b2b-commerce-on-lightning-quickstart#23]
Hi,

I'm trying to implement Stripe Integration for Lightning B2B Commerce Checkout flow.
The provided example of StripeAdapter tries to query the CardPaymentMethod in createAuthResponse() using Id from AuthApiPaymentMethodRequest to get the GatewayToken and assign it as a 'source' to create a Charge. However, AuthApiPaymentMethodRequest has blank Id, and there are no records of type CardPaymentMethod inserted during the checkout flow.
Can you please explain what I'm missing here? At what point in time do we get a Token from Stripe and when do we create a CardPaymentMethod records with this info saved in the GatewayToken field?

**Comment by brettgil** [comment]
Any update here? I am running into this same / similar issue with the CardPaymentMethod query being blank.
During the checkout, the Credit Card component in the Payment and Billing Address subflow just returns a generic front end error: "Gateway call returned an exception. See response for more details." 

Salesforce Customer Support isn't supporting this code and Stripe doesn't support it.

**Comment by brettgil** [comment]
This is an update from Salesforce Support: 
"I think their initial approach was to leverage the StripeAdatper class with the OOTB form; however, the createAuthResponse method checks for a CardPaymentMethod record which errored out the call. The next limitation occurred where Stripe seems to only accept a token or stored payment method for the Auth call, so they needed to get a token prior to making the Auth call. They were unable to achieve that on the OOTB form, so they moved to the Reference LWCs. They were able to get it up and working, but I believe they switched to the iFrame approach after some concerns."

TL;DR - The provide Stripe Payment Gateway code in this repo is only part of the code required.  Additional custom development is required to create a Payment Gateway Adaptor with Payment Methods.  AKA - this is incomplete code.

**Comment by eneag-sf** [comment]
Hello @brettgil, how did you solve this? Would appreciate if you could share with me on how you went with this as I have the same issue. Thanks in advance

**Comment by brettgil** [comment]
Nope - We are looking into a new stripe flow component that recently came out - pay anywhere. There was a beta for this at the end of the year and might be GA now. Contact your Stripe rep.

**Comment by eneag-sf** [comment]
@brettgil Thank you for the reply. I will align with my client and contact together the Stripe rep.
Is there a github repo (or a package) which has this stripe flow component?

**Comment by rubencp** [comment]
I managed to fetch the token and then create the authorization on the same call.
however in case of error I get allways the same Generic error disregarding on what I put on 
`authResponse.setGatewayMessage((String) error.get('message'));`

any ideas?
